xdem/spatialstats.py

```diff
@@ -1067,6 +1298,179 @@ def variogram_sum(h, *args):
+                                          stable_mask: np.ndarray | VectorType | gpd.GeoDataFrame = None,
+                                          unstable_mask: np.ndarray | VectorType | gpd.GeoDataFrame = None,
+                                          errors: np.ndarray | RasterType = None,
+                                          estimator = 'dowd', gsd: float = None, coords: np.ndarray = None,
+                                          subsample: int = 1000, subsample_method: str = 'cdist_equidistant',
+                                          n_variograms: int = 1, n_jobs: int = 1, verbose = False,
+                                          random_state: None | np.random.RandomState | np.random.Generator | int = None,
+                                          bounds: list[tuple[float, float]] = None, p0: list[float] = None,
+                                          **kwargs
+                                          ) -> tuple[pd.DataFrame, pd.DataFrame, Callable[[np.ndarray], np.ndarray]]:
+    """
+    Infer spatial correlation of errors from differenced values on stable terrain and a list of variogram model to fit
+    as a sum.
+
+    This function returns a dataframe of the empirical variogram, a dataframe of optimized model parameters, and a
+    spatial correlation function.
+    It is a convenience wrapper for `estimate_model_spatial_correlation` to work on either Raster or array and compute
+    the stable mask.
+
+    If no stable or unstable mask is provided to mask in or out the values, all terrain is used.
+
+    :param dvalues: Proxy values as array or Raster
+    :param list_models: List of K variogram models to sum for the fit in order from short to long ranges. Can either be
+        a 3-letter string, full string of the variogram name or SciKit-GStat model function (e.g., for a
+        spherical model "Sph", "Spherical" or skgstat.models.spherical).
+    :param stable_mask: Vector shapefile of stable terrain (if dvalues is Raster), or boolean array of same shape as dvalues
+    :param unstable_mask: Vector shapefile of unstable terrain (if dvalues is Raster), or boolean array of same shape as dvalues
+    :param errors: Error values to account for heteroscedasticity (ignored if None).
+    :param estimator: Estimator for the empirical variogram; default to Dowd's variogram (see skgstat.Variogram for
+        the list of available estimators).
+    :param gsd: Ground sampling distance
+    :param coords: Coordinates
+    :param subsample: Number of samples to randomly draw from the values
+    :param subsample_method: Spatial subsampling method
+    :param n_variograms: Number of independent empirical variogram estimations
+    :param n_jobs: Number of processing cores
+    :param verbose: Print statements during processing
+    :param random_state: Random state or seed number to use for calculations (to fix random sampling during testing)
```

Comment:
I'm just thinking. Is it common, like for the functions, to start the argument name with an underscore to state that it is not meant to be used by users?

Reply:
In any case, I would put it at last.